src/ngraph/graph_util.cpp

```diff
@@ -110,6 +110,50 @@ void ngraph::traverse_nodes(const NodeVector& io_nodes,
     }
 }
 
+void ngraph::traverse_nodes(const NodeVector& subgraph_results,
+                            const NodeVector& subgraph_params,
```

Comment:
This version of `traverse_nodes` is mostly copy/paste from the existing (above) version.  Why not just add `subgraph_params` or "stop nodes" as I call them as a default argument to the existing version of `traverse_nodes`.  It does not have to be a default parameter --- this just simplifies the API, no need to change other call sites.